### ISSUE TITLE
Unable to reproduce the same results as given by already trained network

### ISSUE THREAD

**Issue by saxenam06** [mahaitongdae/Approximate-dynamic-Programming#2]
Hello, Many Thanks for this repository. Though I am done understanding all the codes, I am unable to produce the same tracking (trajectory & heading angle error) results as given by the already trained network when I retrain the networks with the given existing hyperparameter settings. I tried multiple iterations on Norm matrix (in Network.py) and  learning rate, however I fail to get the same performance as produced by the already trained network. The trajectory & heading angle error sometimes get very oscillatory with some settings where as for others the peak to peak error increases as compared to the MPC. The value loss converges to 1.e-5 as compared to 1.e-9 (by your trained networks). It would be really helpful if you please help me with some directions on the possible ways I can retrain the networks to get the same performance.  Many Thanks again.

**Comment by mahaitongdae** [comment]
Sorry for the late reply. I also have experienced these issues. Some comments that might be helpful:

- The learning rates are the most effective. Try to adapt the learning rates or the schedules.
- For better simulation performance, Try to adapt the initial distribution in `initialize_state` in [https://github.com/mahaitongdae/Approximate-Dynamic-Programming/blob/demo/dynamics.py](url)  with lower std. This might result in that the task being ''easier'', but the policies might be less robust.

**Comment by saxenam06** [comment]
Thanks for the inputs. I was able to achieve good tracking performance by modifying learning rate and also the utility function. I have a question, When we are benchmarking ADP w.r.t MPC, as per your experience is it not ideal to fix the same weights and have exactly same utility function for both the methods for correlation? I noticed that the utility function weights were different between MPC and ADP. How much correct it is to get the MPC like performance from ADP methods but using a utility function with higher weights? Thanks again!

**Comment by Possiblexhy** [comment]
> Thanks for the inputs. I was able to achieve good tracking performance by modifying learning rate and also the utility function. I have a question, When we are benchmarking ADP w.r.t MPC, as per your experience is it not ideal to fix the same weights and have exactly same utility function for both the methods for correlation? I noticed that the utility function weights were different between MPC and ADP. How much correct it is to get the MPC like performance from ADP methods but using a utility function with higher weights? Thanks again!

Hello, can you share the two neural network learning rate values? I have made many changes, but the results have not been satisfactory.

**Comment by saxenam06** [comment]
Hi @Possiblexhy, please refer to my fork of this repo below where you can find the details of the learning rates, even already trained models and the plots/results  of the tracking performance too! https://github.com/saxenam06/Approximate-Dynamic-Programming

**Comment by Possiblexhy** [comment]
> Hi @Possiblexhy, please refer to my fork of this repo below where you can find the details of the learning rates, even already trained models and the plots/results of the tracking performance too! https://github.com/saxenam06/Approximate-Dynamic-Programming

Thank you very much for your reply!
